ouroboros-consensus/test-consensus/Test/Dynamic/General.hs

```diff
@@ -170,31 +173,70 @@ prop_general ::
      ( Condense blk
      , Eq blk
      , HasHeader blk
+     , RunNode blk
      )
   => SecurityParam
   -> TestConfig
   -> LeaderSchedule
   -> TestOutput blk
   -> Property
 prop_general k TestConfig{numSlots, nodeJoinPlan, nodeTopology} schedule
-  TestOutput{testOutputNodes} =
-    counterexample ("nodeJoinPlan: " <> condense nodeJoinPlan) $
-    counterexample ("schedule: " <> condense schedule) $
+  TestOutput{testOutputNodes, testOutputSlotBlockNos} =
     counterexample ("nodeChains: " <> unlines ("" : map (\x -> "  " <> condense x) (Map.toList nodeChains))) $
+    counterexample ("nodeJoinPlan: " <> condense nodeJoinPlan) $
     counterexample ("nodeTopology: " <> condense nodeTopology) $
+    counterexample ("slot-node-depth: " <> condense slotNodeDepths) $
+    counterexample ("schedule: " <> condense schedule) $
+    counterexample ("schedule': " <> condense schedule') $
```

Comment:
Hmm, we should be able to come up with a clearer name than `schedule'` (I know it's one of the hardest problems :slightly_smiling_face:).

Reply:
I went with `growthSchedule`. I'm tempted to write `actualSchedule`, but there are some corner cases that make that not quite true. I'll keep my eye out for an opportunity to simplify this as I continue with more PRs in this area.